app/src/main/res/layout/view_story_item_footer.xml

```diff
@@ -4,56 +4,60 @@
   xmlns:tools="http://schemas.android.com/tools"
   style="@style/HNews.FeedFooter"
   android:layout_width="match_parent"
-  android:layout_height="@dimen/button_root_height"
-  android:orientation="horizontal">
+  android:layout_height="@dimen/button_root_height">
 
   <ImageButton
     android:id="@+id/article_share_action"
+    android:layout_alignParentLeft="true"
+    android:layout_alignParentStart="true"
     style="@style/HNews.FeedAction"
-    android:src="@drawable/ic_share" />
+    android:src="@drawable/ic_share_grey600_24dp" />
 
   <ImageButton
     android:id="@+id/article_bookmark_action"
+    android:layout_toRightOf="@id/article_share_action"
+    android:layout_toEndOf="@id/article_share_action"
     style="@style/HNews.FeedAction"
     android:src="@drawable/selector_bookmark" />
 
-  <LinearLayout
+  <RelativeLayout
     android:id="@+id/article_comments"
     android:layout_width="wrap_content"
     android:layout_height="match_parent"
+    android:layout_toRightOf="@id/article_bookmark_action"
+    android:layout_toEndOf="@id/article_bookmark_action"
+    android:layout_toStartOf="@id/article_external_action"
+    android:layout_toLeftOf="@id/article_external_action"
```

Comment:
you want to avoid things like this, a `LinearLayout` allows to create your layout based on the orientation and the hierarchy